Jetpack Cloud: fix inconsistencies between modals

rcanepa opened the pull request:
#### Changes proposed in this Pull Request

* Make the buttons sections look the same (modal width, background color, buttons positioning, labels)
* Make the threats container scrollable

#### Implementation notes

- Now the `<Dialog/>` manages how buttons are rendered (we pass them as a prop to the component as we do in the other modals).

#### Questions

- Is the scrollable container a good pattern for this? If yes, how can we make it obvious that it's scrollable? Currently, it doesn't look like it is.

#### Testing instructions

* Go to `http://jetpack.cloud.localhost:3000/`
* Select a site with as many threats you have [1]
* Go to Scan -> Scanner
* Click the `[Fix all]` button
* On another tab, go to the same section a click on any threat the `[Fix threat]` button
* Verify they both look the same (modal width, background color, buttons positioning, and labels)
* Verify the threats section is scrollable (verify this on mobile as well)

[1] You can create a JN site and add threats to it by simply modifying core files. A comment is enough to be marked as a threat.

1151678672052943-1173451195817817

#### Demo

##### Fix threat dialog (this one hasn't changed, it's just for comparison purposes)
![image](https://user-images.githubusercontent.com/3418513/81597667-7415fa00-939c-11ea-8063-749176acf9b5.png)

##### Before
![image](https://user-images.githubusercontent.com/3418513/81597368-04077400-939c-11ea-81a2-7be6ba6ca7d9.png)

##### After
![image](https://user-images.githubusercontent.com/3418513/81597534-4466f200-939c-11ea-8eeb-934359bc41e7.png)

enejb commented:
I tested this and I think it is already an improvement. Nicely done

But I noticed the following issues:
1. There is a horizontal scrolling on mobile for the fix all threats modal 
See
<img width="448" alt="Screen Shot 2020-05-12 at 11 29 25 AM" src="https://user-images.githubusercontent.com/115071/81667878-281e9000-9444-11ea-872b-76778947e372.png">
This horizonat scrolling doesn't exits on the "fit treat modal"

2. Pressing the "Esc" key as well as clicking outside of the "Fix all threats modal" does do anything. The "Fix threat modal" works as expected for me. 

I think the 2 issues can be addressed in a different PR or here.

rcanepa commented:
I love that idea @keoshi ! It's simple enough to implement so I'll add it right here. Thanks! 🙇

rcanepa commented:
Instead of adding what @keoshi proposed on every case, I added it only when the user has more than 3 threats. This is how it looks like in each case:

**3 threats or less**
![image](https://user-images.githubusercontent.com/3418513/81732053-41383880-9466-11ea-9202-538388064b65.png)

**More than 3 threats**
![image](https://user-images.githubusercontent.com/3418513/81732083-4dbc9100-9466-11ea-8437-045ea7e715f8.png)

I also removed the horizontal scrolling that @enejb found.

keoshi commented:
That is awesome, @rcanepa!! 🙌

enejb commented:
@rcanepa 
I pushed a couple of minor changes. 
This is how it look like now. 

<img width="837" alt="Screen Shot 2020-05-13 at 2 38 24 PM" src="https://user-images.githubusercontent.com/115071/81813204-78bae980-9527-11ea-9de1-4b18b99b841d.png">

<img width="807" alt="Screen Shot 2020-05-13 at 2 38 18 PM" src="https://user-images.githubusercontent.com/115071/81813218-7a84ad00-9527-11ea-8bc0-5d427e77806e.png">

keoshi commented:
Nice catch, @enejb ! Thank you both. 🙌
